### PR TITLE
Adding GCC build capability

### PR DESCRIPTION
**Pull request by ggfan**
removing C99 style structure initialization so gcc could compile. mainly on Windows, gcc is the only compiler at this short moment.

### CODE REVIEW

File: tutorial01_load_vulkan/app/src/main/jni/main.cpp [hunk 1]
```diff
@@ -76,15 +76,14 @@ bool initialize(android_app* app) {
...
-      .pNext = nullptr,
-      .apiVersion = VK_MAKE_VERSION(1, 0, 0),
-      .applicationVersion = VK_MAKE_VERSION(1, 0, 0),
-      .engineVersion = VK_MAKE_VERSION(1, 0, 0),
-      .pApplicationName = "tutorial01_load_vulkan",
-      .pEngineName = "tutorial",
-  };
+  VkApplicationInfo appInfo;
+  memset(&appInfo, 0, sizeof(appInfo));
+  appInfo.sType = VK_STRUCTURE_TYPE_APPLICATION_INFO;
```
Comment (Whiteseeker) [inline tutorial01_load_vulkan/app/src/main/jni/main.cpp]: Can't we do:
VkApplicationInfo appInfo = {
       VK_STRUCTURE_TYPE_APPLICATION_INFO,
       nullptr,
[...]

instead ?
This syntax is super verbose...

**Comment by ggfan** [comment]
The thought was that: the structures are big: most are list of values, if we do it this { 1, b, c, d,}, it would be hard from code to know what is what; also some burden on coding: I really have to go through every structure and put the values in the right sequence.

code readability is great with C99 style, but gcc could not take it, it is strange situation ( because clang could not build on windows)

**Comment by Whiteseeker** [comment]
I dunno what to say. If we change for a super verbose notation, knowing
that most vulkan code is filling structure, we will loose a lot of
readability.

**Comment by ggfan** [comment]
I think another way is to keep this windows branch temporarily: when clang is fixed on windows ( in about 2 weeks), this branch could be retired. it is short term thing anyhow, so I think that is fair too.  just change ndk document to point to these 2 branches.

**Comment by ggfan** [comment]
withdraw the pull and keep 2 branches for the time being

**Pull request closed without merging**